app/jobs/clean_deleted_subject_job.rb

```diff
@@ -0,0 +1,15 @@
+# frozen_string_literal: true
+class CleanDeletedSubjectJob < ApplicationJob
+  def perform(subject_id)
+    adapter = Valkyrie::MetadataAdapter.find(:indexing_persister)
+    qs = adapter.query_service
+    folders = qs.find_inverse_references_by(property: :subject, id: subject_id)
+    csp = ChangeSetPersister.new(metadata_adapter: adapter, storage_adapter: Valkyrie.config.storage_adapter)
+    folders.each do |folder|
+      cs = DynamicChangeSet.new(folder).prepopulate!
+      subjects = cs.subject.reject { |id| id.to_s == subject_id }
+      cs.validate(subject: subjects)
+      csp.save(change_set: cs) if cs.valid?
```

Comment:
Do you want to log an error if it's not valid?

Reply:
sure good idea